slm_porosity/heat_transfer.f90

```diff
@@ -826,4 +803,92 @@ FUNCTION Spline_cubic (r_p, l_p, fr_p, fl_p, dfr_p, dfl_p)
+    IF (.NOT.PRESENT(is_D).OR.(.NOT.is_D)) THEN
+      WHERE (left_p.LT.field .AND. field.LT.right_p)
+        piecewise_connection = (v_right - v_left)*der*field + (right_p*v_left - left_p*v_right)*der
+      ELSEWHERE (field.LE.left_p)
+        piecewise_connection = a1*field + b1
+      ELSEWHERE (right_p.LE.field)
+        piecewise_connection = a2*field + b2
+      END WHERE
+    ELSE
+      WHERE (left_p.LT.field .AND. field.LT.right_p)
+        piecewise_connection = (v_right - v_left)*der
+      ELSEWHERE (field.LE.left_p)
+        piecewise_connection = a1
+      ELSEWHERE (right_p.LE.field)
+        piecewise_connection = a2
+      END WHERE
+    END IF
+  END FUNCTION piecewise_connection
+
+  !spline smoothness of two lines a1*x+b1 and a2*x+b2 with the drop declared by two points: left_p and right_p
+  FUNCTION cubic_splines_smooth (field, left_p, right_p, a1, a2, b1, b2, is_D)
+    IMPLICIT NONE
+    REAL (pr),         INTENT(IN) :: field(:), left_p, right_p, a1, a2, b1, b2
+    LOGICAL, OPTIONAL, INTENT(IN) :: is_D
+    REAL (pr) :: cubic_splines_smooth(SIZE(field))
+    REAL (pr), DIMENSION(4) :: coeff_left, coeff_right
+    REAL (pr) :: der, v_left, v_right, v_drop_left, v_drop_right, smooth_delta
+    REAL (pr) :: leftm, leftp, rightm, rightp
+    smooth_delta = (right_p - left_p)*smoothing_width              ! width of the cubic splines
+    der = 1.0_pr/(right_p - left_p)                                       ! first derivative
+    leftm = left_p - smooth_delta
+    leftp = left_p + smooth_delta
+    rightm = right_p - smooth_delta
+    rightp = right_p + smooth_delta
+    v_left = a1*left_p + b1
+    v_right = a2*right_p + b2
+    v_drop_left = (v_right - v_left)*der*leftp + (right_p*v_left - left_p*v_right)*der
+    v_drop_right = (v_right - v_left)*der*rightm + (right_p*v_left - left_p*v_right)*der
+    coeff_left = Spline_cubic(leftm, leftp, a1*leftm+b1, v_drop_left, a1, (v_right - v_left)*der)
+    coeff_right = Spline_cubic(rightm, rightp, v_drop_right, a2*rightp+b2, (v_right - v_left)*der, a2)
+    cubic_splines_smooth = piecewise_connection(field, left_p, right_p, a1, a2, b1, b2, is_D)
+
+    IF (.NOT.PRESENT(is_D).OR.(.NOT.is_D)) THEN
+      WHERE (field.GT.leftm .AND. field.LE.leftp)
+        cubic_splines_smooth = coeff_left(1)*field**3 + coeff_left(2)*field**2 &
+          + coeff_left(3)*field + coeff_left(4)
+      ELSEWHERE (field.GT.rightm .AND. field.LE.rightp)
+        cubic_splines_smooth = coeff_right(1)*field**3 + coeff_right(2)*field**2 &
+          + coeff_right(3)*field + coeff_right(4)
+      END WHERE
+    ELSE
+      WHERE (field.GT.leftm .AND. field.LE.leftp)
+        cubic_splines_smooth = 3*coeff_left(1)*field**2 + 2*coeff_left(2)*field + coeff_left(3)
+      ELSEWHERE (field.GT.rightm .AND. field.LE.rightp)
+        cubic_splines_smooth = 3*coeff_right(1)*field**2 + 2*coeff_right(2)*field + coeff_right(3)
+      END WHERE
+    END IF
+  END FUNCTION cubic_splines_smooth
```

Comment:
All this stuff should be removed in favor of the old version. Let's do all via `liquid_fraction`. Use its symmetry.